application/src/command.rs

```diff
@@ -0,0 +1,55 @@
+use crate::port::{TodoListRepository, TodoListStore};
+use domain::todolist_message::TodoListMessage;
+use framework::*;
+use serde::Deserialize;
+
+#[derive(Deserialize)]
+pub enum Command {
+    AddTask { name: String },
+    RemoveTask { index: usize },
+    CompleteTask { index: usize },
+}
+
+#[derive(Debug, Error)]
+pub enum CommandError {
+    #[error("Task name cannot be empty")]
+    TaskNameCannotBeEmpty,
+}
+
+impl TryInto<TodoListMessage> for &Command {
+    type Error = CommandError;
+
+    fn try_into(self) -> Result<TodoListMessage, Self::Error> {
+        Ok(match self {
+            Command::AddTask { name } => TodoListMessage::AddTask(
+                name.clone()
+                    .try_into()
+                    .map_err(|_| CommandError::TaskNameCannotBeEmpty)?,
+            ),
+            Command::RemoveTask { index } => TodoListMessage::RemoveTask((*index).into()),
+            Command::CompleteTask { index } => TodoListMessage::CompleteTask((*index).into()),
+        })
+    }
+}
```

Comment:
**Suggestion:** Consider adding a docstring to the `try_into` function. [best practice]
```suggestion
impl TryInto<TodoListMessage> for &Command {
  /// Tries to convert the command into a `TodoListMessage`.
  /// Returns an error if the conversion fails.
  type Error = CommandError;

  fn try_into(self) -> Result<TodoListMessage, Self::Error> {
...
```